src/core/utils/helpers/lazy-load.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef, useState } from "react";
+import { useIntersection } from "react-use";
+
+export const useListItemLazyLoad = () => {
+  const listItemRef = useRef(null);
+  const intersection = useIntersection(listItemRef, {
+    root: null,
+    rootMargin: "0%",
+    threshold: 0.33
```

Comment:
Consider adding an inline comment explaining why you ended up with 0.33.

I can figure out why we do not use 1 (ever pixel) but why not instead of say 0 which would run if even 1 pixel is visible).

Reply:
We go with 0. One could consider even use a negative value but let's stick to 0!